lib/statsd/instrument/batched_udp_sink.rb

```diff
@@ -80,11 +83,12 @@ def flush
       end
 
       def dispatch
-        loop do
+        until @interrupted
```

Comment:
Alternatively, we could use `pending_interrupt?`/`handle_interrupt` and `Thread.raise` for exiting the thread.

Reply:
Nah, this is fine. I'd rather avoid `Thread.raise`.

Reply:
Hum, actually wait. if we are interrupted we should still flush the queue.